File: solution/2966. Divide Array Into Arrays With Max Difference/2966.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        
        vector<vector<int>> res;
        res.push_back({});
        for (int x : nums) {
            if (res.back().empty()) {
                res.back().push_back(x);
                continue;
            }
            if (res.back().size() < 3 && x - res.back()[0] <= k) {
                res.back().push_back(x);
            } else {
                if (res.back().size() < 3) return {};
                res.push_back({x});
            }
        }
        if (res.back().size() < 3) return {};
        return res;
    }
};

File: solution/2966. Divide Array Into Arrays With Max Difference/2966.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        res = []
        nums.sort()
        for i in range(2, len(nums), 3):
            if nums[i] - nums[i - 2] > k:
                return []
            res.append([nums[i - 2], nums[i - 1], nums[i]])
        return res
    
